### PR TITLE
Metrics service update

### PR DESCRIPTION
**Pull request by flicker-harmony**
Added Tx Pool size metric (gauge type), added -metrics flag to node.sh (false by default - doesn't collect metrics), fixed storage (dumps metrics if failed to push to prometheus).

### CODE REVIEW

**Comment by harmony-ek** [comment]
Jenkins speaking; can one of the admins verify this patch?

**Comment by LeoHChen** [comment]
Jenkins, test this please.

File: core/tx_pool.go [hunk 1]
```diff
@@ -463,6 +463,11 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 	pool.promoteExecutables(nil)
 }
 
+// GetTxPoolSize returns tx pool size.
+func (pool *TxPool) GetTxPoolSize() uint64 {
```
Comment (denniswon) [inline core/tx_pool.go]: is this class actually used? I thought this was not being used yet until txPool integration happens.

File: internal/configs/node/config.go [hunk 2]
```diff
@@ -178,6 +179,16 @@ func (conf *ConfigType) SetPushgatewayPort(port string) {
 	conf.PushgatewayPort = port
 }
 
+// SetMetricsFlag set the metrics flag
```
Comment (denniswon) [inline internal/configs/node/config.go]: @LeoHChen I'm a bit confused whether nodeConfig is being deprecated or not.
Maybe I misheard, but thought shardingconfig is where new configurations should be added (which also made it confusing if configs unrelated to sharding specifically should also be put there)

Could you clarify?

File: core/tx_pool.go (hunk 1 above)
Comment (flicker-harmony) [reply core/tx_pool.go]: It is used for tx pool size metric.

Comment (denniswon) [reply core/tx_pool.go]: but we are not using TxPool yet. cc: @rlan35 @mikedoan ?

File: internal/configs/node/config.go (hunk 2 above)
Comment (LeoHChen) [reply internal/configs/node/config.go]: nodeConfig is mostly used when we have command-line flags.  shardingconfig is not for command-line, those are built-in parameters that shouldn't be changed in command line.

**Comment by denniswon** [comment]
Tested:
Confirm master unbroken with this pr.

```
dennis.won@mp-dennisw-ml2:~/harmony/harmony (pr_metrics_service_update) $ ./bin/wallet -p local transfer --from one103q7qe5t2505lypvltkqtddaef5tzfxwsse4z7 --to one1pdv9lrdwl0rg5vglh4xtyrv3wjk3wsqket7zxy --amount 40 --shardID 0
Using local profile for wallet
Unlock account succeeded! ''

Transaction Id for shard 0: 0xee147c7163255b01fb059f365d9b5efb4b4ed4b1b8306eec3a323ed0a0fe4880
```

cc: @flicker-harmony @LeoHChen @xChickens

**Comment by xChickens** [comment]
brave_xHSW3NH3Sc
MobaXterm_i6ivTJUxAX

**Comment by harmony-ek** [comment]
@flicker-harmony, next time you modify `scripts/node.sh` loop me in please.  I needed to fix some bugs.  This scripts is externally used by all of our node operators, so we must be extra careful.  Cc: @LeoHChen

**Pull request merged**